### PR TITLE
Backporting release-v2.8 into dev-v2.8 to pull in feb patches

### PR DESCRIPTION
**Pull request by thatmidwesterncoder**
For https://github.com/rancher/rancher/issues/44520

### CODE REVIEW

File: channels-rke2.yaml [hunk 1]
```diff
@@ -1731,3 +1778,50 @@ releases:
     charts: &charts-v1-28-6-rke2r1
       <<: *charts-v1-27-10-rke2r1
     featureVersions: *featureVersions-v1
+  - version: v1.27.11+rke2r1
```
Comment (snasovich) [inline channels-rke2.yaml:1765]: This needs to be added before `- version: v1.28.6+rke2r1` (which itself needs to be updated - more on this in a different comment) for the proper order of versions.

File: pkg/rke/k8s_rke_system_images.go [hunk 2]
```diff
@@ -10000,12 +10126,12 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
-			AciOpflexContainer:        "noiro/opflex:6.0.3.2.81c2369",
-			AciMcastContainer:         "noiro/opflex:6.0.3.2.81c2369",
-			AciOpenvSwitchContainer:   "noiro/openvswitch:6.0.3.2.81c2369",
-			AciControllerContainer:    "noiro/aci-containers-controller:6.0.3.2.81c2369",
+			AciCniDeployContainer:     "noiro/cnideploy:6.0.4.1.81c2369",
+			AciHostContainer:          "noiro/aci-containers-host:6.0.4.1.81c2369",
+			AciOpflexContainer:        "noiro/opflex:6.0.4.1.81c2369",
+			AciMcastContainer:         "noiro/opflex:6.0.4.1.81c2369",
+			AciOpenvSwitchContainer:   "noiro/openvswitch:6.0.4.1.81c2369",
+			AciControllerContainer:    "noiro/aci-containers-controller:6.0.4.1.81c2369",
```
Comment (snasovich) [inline pkg/rke/k8s_rke_system_images.go:10134]: We want the same versions of `Aci...` images defined here be also used for `v1.28.7-rancher1-1` defined below.
(Note - this comment is not to update the lines the comment is on - it's just the best place to leave the comment on as lines for `v1.28.7-rancher1-1` were not modified and I cannot leave a comment there)

File: channels-rke2.yaml [hunk 3]
```diff
@@ -1731,3 +1778,50 @@ releases:
     charts: &charts-v1-28-6-rke2r1
```
Comment (snasovich) [inline channels-rke2.yaml]: Again, this comment is not for this immediate line but GH PR review limitations...

We want `v1.28.6+rke2r1` replaced with `v1.28.7+rke2r1`, including making the changes in its definition to define proper `serverArgs` / `agentArgs` / `charts` that account for additional options that were added in Feb patches (like an updated list of CNIs defined in `v1.27.11+rke2r1` that are also supported in `v1.28.7+rke2r1`) + new args that are supported in 1.28.x only (like `embedded-registry` server arg already added in `v1.28.6+rke2r1`).

Will likely need help from RKE2 team (@rancher-max / @briandowns ) to sort out through this.

File: channels.yaml [hunk 4]
```diff
@@ -550,3 +556,9 @@ releases:
       disable-default-registry-endpoint:
         type: boolean
     featureVersions: *featureVersions-v1
+  - version: v1.27.11+k3s1
```
Comment (snasovich) [inline channels.yaml]: Same comment as for RKE2 versions like putting `1.27.11+k3s1` above 1.28.x definition and changing `v1.28.6+k3s2` to `v1.28.7+k3s1`.

**Review by snasovich — CHANGES_REQUESTED** [review]
@thatmidwesterncoder , thank you for opening this issue. I've linked it to https://github.com/rancher/rancher/issues/44520 as it's a relevant issue.

Now, this "backport" is a bit more involved due to the presence of `1.28.x` versions in `dev-v2.8` for anticipated support of 1.28 in the upcoming 2.8.x release.
Luckily for RKE1 case it's mostly taken care of by https://github.com/rancher/kontainer-driver-metadata/pull/1337 but there is still one minor change to do in regards to ACI CNI (left a comment on that).

The situation with RKE2 is a bit more involved and may require some help from RKE2 team (also left specific comments on that).

File: channels-rke2.yaml [hunk 5]
```diff
@@ -1715,19 +1762,66 @@ releases:
...
+    agentArgs: &agentArgs-v1-28-7-rke2r1
       <<: *agentArgs-v1-25-15-rke2r2
       default-runtime:
         type: string
       disable-default-registry-endpoint:
         type: boolean
-    charts: &charts-v1-28-6-rke2r1
+    charts: &charts-v1-28-7-rke2r1
       <<: *charts-v1-27-10-rke2r1
     featureVersions: *featureVersions-v1
```
Comment (snasovich) [inline channels-rke2.yaml:1827]: Follow-up to now-resolved https://github.com/rancher/kontainer-driver-metadata/pull/1354#discussion_r1517912525

Certainly need RKE2 team input (@briandowns) on this but I gather (again, not sure below is correct until we get input from RKE2 team):
1. For `serverargs` we will be missing updated `cni` options that are supported in 1.28.7 same as in v2.7.11. So we _should_ be able to just change anchor to `*serverArgs-v1-27-11-rke2r1`.
2. `agentargs` LGTM
3. `charts` likely will be the same as on 1.27.11 as well so change anchor to `*charts-v1-27-11-rke2r1`
4. `featureVersions` LGTM
5. Not sure if there are any additional args/charts on 1.28.6 -> 1.28.7

So, something like:
```
  - version: v1.28.7+rke2r1
    minChannelServerVersion: v2.8.3-alpha1
    maxChannelServerVersion: v2.8.99
    serverArgs: &serverArgs-v1-28-7-rke2r1
      <<: *serverArgs-v1-27-11-rke2r1
      embedded-registry:
        type: boolean
    agentArgs: &agentArgs-v1-28-7-rke2r1
      <<: *agentArgs-v1-25-15-rke2r2
      default-runtime:
        type: string
      disable-default-registry-endpoint:
        type: boolean
    charts: &charts-v1-28-7-rke2r1
      <<: *charts-v1-27-11-rke2r1
    featureVersions: *featureVersions-v1
```

**Review by snasovich — CHANGES_REQUESTED** [review]
Almost LGTM, one comment to address

File: channels-rke2.yaml (hunk 5 above)
Comment (briandowns) [reply channels-rke2.yaml:1827]: Server args referencing v1.27.11 is fine. The charts ref should be fine as well.

CNI addition to server args:

```yaml
      cni:
        default: calico
        type: array
        options:
        - canal
        - cilium
        - calico
        - flannel
        - multus,canal
        - multus,cilium
        - multus,calico
```

Comment (snasovich) [reply channels-rke2.yaml:1827]: @briandowns , just confirming you're fine with the approach of just referencing `serverArgs-v1-27-11-rke2r1` and just adding `embedded-registry` to it. The reason I'm a bit confused as you stated "CNI addition to server args:" with YAML following it but if we reference `serverArgs-v1-27-11-rke2r1` we will get CNI addition taken care of it I believe. Or am I missing something?

Comment (briandowns) [reply channels-rke2.yaml:1827]: My fault, it'll get brought in from that tag.

**Comment by jiaqiluo** [comment]
replaced by https://github.com/rancher/kontainer-driver-metadata/pull/1356

**Pull request closed without merging**